### PR TITLE
scalability framework: fix name handling

### PR DESCRIPTION
**Pull request by nrainer-materialize**
This addresses the issue `assert self._port is not None` reported by @danhhz in https://materializeinc.slack.com/archives/C01LKF361MZ/p1698273931269339 and observed in [Nightly.](https://buildkite.com/materialize/nightlies/builds/4888#018b692d-3557-4354-847a-a2ec385a6c6f)

### Commits
* 815765a1c0 scalability framework: fix
* 71bf057442 scalability framework: rework endpoint name

### CODE REVIEW

File: misc/python/materialize/scalability/endpoint.py [hunk 1]
```diff
@@ -46,13 +46,21 @@ def sql(self, sql: str) -> None:
         cursor = conn.cursor()
         cursor.execute(sql.encode("utf8"))
 
-    def name(self) -> str:
-        if self._name is not None:
-            return self._name
+    def try_load_version(self) -> str:
+        """
+        Tries to load the version from the database or returns 'unknown' otherwise.
```
Comment (philip-stoev) [inline misc/python/materialize/scalability/endpoint.py:51]: Under what circumstances would `unknown` be returned?

Comment (nrainer-materialize) [reply misc/python/materialize/scalability/endpoint.py:51]: The problem is that you can't invoke this method at any time. That's also what caused the error. The endpoint needs to be up for this to work. That is, if you invoke this method too early (before having executed `endpoint.up()`), the query will fail.

Comment (philip-stoev) [reply misc/python/materialize/scalability/endpoint.py:51]: I see.

Comment (nrainer-materialize) [reply misc/python/materialize/scalability/endpoint.py:51]: The same happened if you invoked it too late on an earlier version when the endpoint was already down. I had fixed this by caching the information once I have it.

**Review by philip-stoev — APPROVED** [review]
I think this will need to be adjusted going forward if too many `unknown` starts showing up in results and such. But as an emergency fix for Nightly that would be ok.

**Pull request merged**